libs/trainer/mlpe/trainer/trainer.py

```diff
@@ -110,14 +110,15 @@ def train(
     max_epochs: int = 40,
     init_weights: Optional[Path] = None,
     lr: float = 1e-3,
-    min_lr: float = 1e-5,
-    decay_steps: int = 10000,
-    weight_decay: float = 0.0,
     early_stop: int = 20,
     # misc params
     device: Optional[str] = None,
     use_amp: bool = False,
     profile: bool = False,
+    optimizer_name: str = "Adam",
```

Comment:
Similar comments as #38. We can annotate this as 

```python
def train(
    optimizer: Callable, 
    optimizer_kwargs: Dict,
    ...
):
    optimizer = optimzer(optimizer_kwargs)
```
and then in the pyproject.toml have

```toml
[tool.typeo.scripts.train]
optimizer = "torch.optimizers.lr_scheduler.Adam"
optimizer_kwargs = {arg1: "arg1"}

Reply:
good idea.

Reply:
done.